packages/usa-modal/src/index.js

```diff
@@ -278,51 +279,50 @@ const cleanUpModal = (baseComponent) => {
   modalWrapper.parentElement.removeChild(modalWrapper);
 };
 
-modal = {
-  init(root) {
-    selectOrMatches(MODAL, root).forEach((modalWindow) => {
-      const modalId = modalWindow.id;
-      setUpModal(modalWindow);
-
-      // this will query all openers and closers including the overlay
-      document
-        .querySelectorAll(`[aria-controls="${modalId}"]`)
-        .forEach((item) => {
-          // Turn anchor links into buttons because of
-          // VoiceOver on Safari
-          if (item.nodeName === "A") {
-            item.setAttribute("role", "button");
-            item.addEventListener("click", (e) => e.preventDefault());
-          }
-
-          // Can uncomment when aria-haspopup="dialog" is supported
-          // https://a11ysupport.io/tech/aria/aria-haspopup_attribute
-          // Most screen readers support aria-haspopup, but might announce
-          // as opening a menu if "dialog" is not supported.
-          // item.setAttribute("aria-haspopup", "dialog");
-
-          item.addEventListener("click", toggleModal);
-        });
-    });
-  },
-  teardown(root) {
-    selectOrMatches(MODAL, root).forEach((modalWindow) => {
-      cleanUpModal(modalWindow);
-      const modalId = modalWindow.id;
-
-      document
-        .querySelectorAll(`[aria-controls="${modalId}"]`)
-        .forEach((item) => item.removeEventListener("click", toggleModal));
-    });
-  },
-  focusTrap: null,
-  toggleModal,
-  on(root) {
-    this.init(root);
-  },
-  off(root) {
-    this.teardown(root);
-  },
-};
+modal = behavior(
+  {},
+  {
+    init(root) {
+      selectOrMatches(MODAL, root).forEach((modalWindow) => {
+        const modalId = modalWindow.id;
+        if (modalId) {
+          setUpModal(modalWindow);
+        }
+
+        // this will query all openers and closers including the overlay
+        document
+          .querySelectorAll(`[aria-controls="${modalId}"]`)
+          .forEach((item) => {
+            // Turn anchor links into buttons because of
+            // VoiceOver on Safari
+            if (item.nodeName === "A") {
+              item.setAttribute("role", "button");
+              item.addEventListener("click", (e) => e.preventDefault());
+            }
+
+            // Can uncomment when aria-haspopup="dialog" is supported
+            // https://a11ysupport.io/tech/aria/aria-haspopup_attribute
+            // Most screen readers support aria-haspopup, but might announce
+            // as opening a menu if "dialog" is not supported.
+            // item.setAttribute("aria-haspopup", "dialog");
+
+            item.addEventListener("click", toggleModal);
+          });
+      });
+    },
+    teardown(root) {
+      selectOrMatches(MODAL, root).forEach((modalWindow) => {
+        cleanUpModal(modalWindow);
+        const modalId = modalWindow.id;
+
+        document
+          .querySelectorAll(`[aria-controls="${modalId}"]`)
```

Comment:
Again, you can use `selectOrMatches` to select and pass `root`, which could be document or something else.

Reply:
Resolved in [1c51556](https://github.com/uswds/uswds/pull/5208/commits/1c515565af6ced9e140fe588cd754d55e483f53b)